fastlane/lib/fastlane/actions/automatic_code_signing.rb

```diff
@@ -40,6 +40,9 @@ def self.run(params)
           end
           if params[:code_sign_identity]
             build_configuration_list.set_setting("CODE_SIGN_IDENTITY", params[:code_sign_identity])
+            build_configuration_list.set_setting("CODE_SIGN_IDENTITY[sdk=*]", params[:code_sign_identity])
+            build_configuration_list.set_setting("CODE_SIGN_IDENTITY[sdk=iphoneos*]", params[:code_sign_identity])
+            build_configuration_list.set_setting("CODE_SIGN_IDENTITY[sdk=appletvos*]", params[:code_sign_identity])
```

Comment:
Is there a way to get/set these dynamically wit `xcodeproj` instead of hardcoding `sdk=*`, `sdk=iphoneos*`, and `sdk=appletvos*`?

Reply:
Hey @joshdholtz , my response by mail went in the bad thread.

Sadly I think it is not. The `build_settings` [method](http://www.rubydoc.info/gems/xcodeproj/Xcodeproj/Project/Object/XCConfigurationList#build_settings-instance_method) of the gem only works for a specific build configuration.
What we could do is 
• Loop on the configurations
• Call build_settings
• Search the ones with `CODE_SIGN_IDENTITY`
• Replace if exists

This seems a bit too much.

Reply:
I _think_ lopping through all configurations might be the correct approach though 🙃 This will make this more stable if/when a new platform is added and _should_ also make it work for Mac apps (which I don't think the current solution would account for with the things being hardcoded?)